libraries/ESP32/examples/MacAddress/GetMacAddress/GetMacAddress.ino

```diff
@@ -0,0 +1,85 @@
+/*
+ * GetMacAddress
+ * 
+ * This sketch prints out the MAC addresses for different interfaces.
+ * 
+ * Written by: Daniel Nebert
+ * 
+ * The first printed MAC address is obtained by calling 'esp_efuse_mac_get_default'
+ * (https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/misc_system_api.html#_CPPv425esp_efuse_mac_get_defaultP7uint8_t)
+ * which returns base MAC address which is factory-programmed by Espressif in EFUSE.
+ * 
+ * The remaining printed MAC addresses is obtained by calling 'esp_read_mac'
+ * (https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/misc_system_api.html#_CPPv412esp_read_macP7uint8_t14esp_mac_type_t)
+ * and passing in the 'esp_mac_type_t' type.
+ * (https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/misc_system_api.html#esp__mac_8h_1a1b19aca597277a2179682869c140477a)
+ * 
+
+enum esp_mac_type_t
+
+  Values:
+  
+  enumerator ESP_MAC_WIFI_STA
+    MAC for WiFi Station (6 bytes)
+  
+  enumerator ESP_MAC_WIFI_SOFTAP
+    MAC for WiFi Soft-AP (6 bytes)
+  
+  enumerator ESP_MAC_BT
+    MAC for Bluetooth (6 bytes)
+  
+  enumerator ESP_MAC_ETH
+    MAC for Ethernet (6 bytes)
+  
+  enumerator ESP_MAC_IEEE802154
+    if CONFIG_SOC_IEEE802154_SUPPORTED=y, MAC for IEEE802154 (8 bytes)
+  
+  enumerator ESP_MAC_BASE
+    Base MAC for that used for other MAC types (6 bytes)
+  
+  enumerator ESP_MAC_EFUSE_FACTORY
+    MAC_FACTORY eFuse which was burned by Espressif in production (6 bytes)
+  
+  enumerator ESP_MAC_EFUSE_CUSTOM
+    MAC_CUSTOM eFuse which was can be burned by customer (6 bytes)
+  
+  enumerator ESP_MAC_EFUSE_EXT
+    if CONFIG_SOC_IEEE802154_SUPPORTED=y, MAC_EXT eFuse which is used as an extender for IEEE802154 MAC (2 bytes)
+
+*/

```

Comment:
Link to `release-v5.1` rather than `latest` as we don't know what might change in the future versions.

Also, IMO you don't need to put the enum in the description if you are providing the link for it.

```suggestion
 * The first printed MAC address is obtained by calling 'esp_efuse_mac_get_default'
 * (https://docs.espressif.com/projects/esp-idf/en/release-v5.1/esp32/api-reference/system/misc_system_api.html#_CPPv425esp_efuse_mac_get_defaultP7uint8_t)
 * which returns base MAC address which is factory-programmed by Espressif in EFUSE.
 * 
 * The remaining printed MAC addresses is obtained by calling 'esp_read_mac'
 * (https://docs.espressif.com/projects/esp-idf/en/release-v5.1/esp32/api-reference/system/misc_system_api.html#_CPPv412esp_read_macP7uint8_t14esp_mac_type_t)
 * and passing in the 'esp_mac_type_t' type.
 * (https://docs.espressif.com/projects/esp-idf/en/release-v5.1/esp32/api-reference/system/misc_system_api.html#esp__mac_8h_1a1b19aca597277a2179682869c140477a)
 * Please check the documentation for all available MAC type options.
 */
```

Reply:
I provided the list, but condensed it down.  Just incase they can't get to the URL for some reason.